src/parser/FStar.Parser.ToDocument.fs

```diff
@@ -1355,21 +1728,26 @@ and p_refinedBinder b phi =
 
 (* A simple def can be followed by a ';'. Protect except for the last one. *)
 and p_simpleDef ps (lid, e) =
-  group (p_qlident lid ^/^ equals ^/^ p_noSeqTerm ps false e)
+  group (p_qlident lid ^/^ equals ^/^ p_noSeqTermAndComment ps false e)
 
 and p_appTerm e = match e.tm with
   | App _ when is_general_application e ->
       let head, args = head_and_args e in
-      let head_doc, args =
-        if !should_print_fs_typ_app
-        then
-          let fs_typ_args, args = BU.take (fun (_,aq) -> aq = FsTypApp) args in
-          p_indexingTerm head ^^
-          soft_surround_map_or_flow 2 0 empty langle (comma ^^ break1) rangle p_fsTypArg fs_typ_args,
-          args
-        else p_indexingTerm head, args
-      in
-      group (soft_surround_map_or_flow 2 0 head_doc (head_doc ^^ space) break1 empty p_argTerm args)
+      (match args with
+      | [e1; e2] when snd e1 = Infix ->
```

Comment:
I've recently allowed to use any term as an infix operator (see #1604), which makes this not work so well. For instance ```1 `F` 3``` goes to the `Construct` branch below, and ```a `b c` d``` is an application of three arguments, so it does not match this branch. I wonder if we should keep an `InfixApp` node instead of relying on this qualifier?